tools/testsys/src/vmware_k8s.rs

```diff
@@ -0,0 +1,242 @@
+            .name(cluster_input.cluster_name)
+            .set_labels(Some(labels))
+            .control_plane_endpoint_ip(control_plane_endpoint)
+            .creation_policy(CreationPolicy::IfNotExists)
+            .version(cluster_version)
+            .ova_name(self.image_id(cluster_input.crd_input)?)
+            .tuf_repo(
+                cluster_input
+                    .crd_input
+                    .tuf_metadata()
+                    .context(error::InvalidSnafu {
+                        what: "TUF repo information is required for vmware cluster creation.",
+                    })?,
+            )
+            .vcenter_host_url(&self.datacenter.vsphere_url)
+            .vcenter_datacenter(&self.datacenter.datacenter)
+            .vcenter_datastore(&self.datacenter.datastore)
+            .vcenter_network(&self.datacenter.network)
+            .vcenter_resource_pool(&self.datacenter.resource_pool)
+            .vcenter_workload_folder(&self.datacenter.folder)
+            .mgmt_cluster_kubeconfig_base64(&self.encoded_mgmt_cluster_kubeconfig)
+            .set_conflicts_with(Some(existing_clusters))
+            .destruction_policy(DestructionPolicy::OnTestSuccess)
+            .image(
+                cluster_input
+                    .crd_input
+                    .images
+                    .vsphere_k8s_cluster_resource_agent_image
+                    .as_ref()
+                    .expect(
+                        "The default VSphere K8s cluster resource provider image URI is missing.",
+                    ),
+            )
+            .set_image_pull_secret(
+                cluster_input
+                    .crd_input
+                    .images
+                    .testsys_agent_pull_secret
+                    .to_owned(),
+            )
+            .set_secrets(Some(cluster_input.crd_input.config.secrets.clone()))
+            .privileged(true)
+            .build(cluster_input.cluster_name)
+            .map_err(|e| error::Error::Build {
+                what: "VSphere k8s cluster CRD".to_string(),
+                error: e.to_string(),
+            })?;
+        Ok(CreateCrdOutput::NewCrd(Box::new(Crd::Resource(
+            vsphere_k8s_crd,
+        ))))
+    }
+
+    /// Create a vSphere VM provider CRD to launch Bottlerocket VMs on the cluster created by
+    /// `cluster_crd`.
+    async fn bottlerocket_crd<'a>(
+        &self,
+        bottlerocket_input: BottlerocketInput<'a>,
+    ) -> Result<CreateCrdOutput> {
+        let cluster_name = bottlerocket_input
+            .cluster_crd_name
+            .as_ref()
+            .expect("A vSphere K8s cluster provider is required");
+        let labels = bottlerocket_input.crd_input.labels(btreemap! {
+            "testsys/type".to_string() => "vms".to_string(),
+            "testsys/cluster".to_string() => cluster_name.to_string(),
+        });
+
+        // Check if other VMs are using this cluster
```

Comment:
I assume "VMs are using this cluster" means the cluster exists. If that is a problem (I assume it is since a few lines below there is a `conflicts_with` statement), it should be documented so that it is obvious to the users that they are going down the wrong path. I don't like going through all the steps to set up the testing infrastructure, get at the end of it and see an error not obvious enough that just says "you can't use cluster <blah>", which will likely require me cleaning up and start over.

Reply:
`conflicts_with` does not denote an error. It is used to tell the controller that this resource should not be created until the `conflicting` resources have been cleaned up.

Reply:
> until the conflicting resources have been cleaned up

Similar to my comment above, will the user have to clean up the conflicting resources? If so, do we provide an obvious enough failure mode that lets them know they have to clean up resources?

Reply:
The conflicting resources are automatically cleaned up on test success.